### ISSUE TITLE
Vendored Sourceror so we can use it in elixir-ls

### ISSUE THREAD

**Issue by robmckinnon** [elixir-lsp/elixir-ls#1007]
The elixir-ls team have said that for `sourceror` to be used in `elixir-ls` they "[need it vendored like dialyxir and jason](https://github.com/elixir-lsp/elixir-ls/pull/841#issuecomment-1589802871)".

When I checked, this means that the `elixir-ls` uses a dep branch that has all [`Dialyxir` modules renamed to `DialyxirVendored`](https://github.com/jeremyjh/dialyxir/commit/b610d1a87885576e0e2aa4cfb5e176072bcd9457).

Would you consider creating a fork of sourceror e.g. `elixir-lsp/sourceror` that mirrors the latest `sourceror` release with all `Sourceror` modules renamed to `VendoredSourceror`?

**Comment by lukaszsamson** [comment]
> Would you consider creating a fork of sourceror

Yes if sourceror is useful. However, currently all experimental code using it has been removed and sourceror is not a dependency

**Comment by robmckinnon** [comment]
Thanks for saying it may be possible. I've got a draft PR I'm working on to add Exract Function refactoring support to elixir-ls, which depends on Sourceror.
https://github.com/elixir-lsp/elixir-ls/pull/878

I'm at CodeBeam Europe now. If you happen to be here too we could discuss in person.
